agent/lib_guzzle_common.h

```diff
@@ -31,16 +31,13 @@ extern char* nr_guzzle_create_async_context_name(const char* prefix,
 extern int nr_guzzle_in_call_stack(TSRMLS_D);
 
 /*
- * Purpose : Checks if the given object implements
- *           GuzzleHttp\Event\HasEmitterInterface. For a Client object, this
- *           indicates that the object is from Guzzle 4 or 5.
+ * Purpose: This function checks which guzzle version is being used by the object
  *
  * Params  : 1. The object to check.
- *
- * Returns : Non-zero if the object does implement the interface; zero
- *           otherwise.
+ * 
+ * Returns : A string indicating the guzzle version being used
```

Comment:
Does this function allocate a new string? Is the caller responsible for freeing the result?

Reply:
Yes, it allocates a new string and the caller is responsible for freeing the returned result. I have updated the description of the function to include that the returned string must be freed. [8065805](https://github.com/newrelic/newrelic-php-agent/pull/498/commits/8065805718d4ab2b4b9bc18d6fb8af5e6730773a)